plugins/guests/arch/cap/configure_networks.rb

```diff
@@ -41,10 +41,9 @@ def self.configure_networks(machine, networks)
 
             commands << <<-EOH.gsub(/^ {14}/, '').rstrip
               # Configure #{network[:device]}
-              mv '#{remote_path}' '/etc/netctl/#{network[:device]}' &&
-              ip link set '#{network[:device]}' down &&
-              netctl restart '#{network[:device]}' &&
-              netctl enable '#{network[:device]}'
+              chmod 0644 '#{remote_path}' &&
+              mv '#{remote_path}' '/etc/systemd/network/#{network[:device]}.network' &&
+              networkctl reload
```

Comment:
I think this might be a pretty destructive change for folks who are using netctl in their boxes. Vagrant could accommodate using either tool by checking what is available on the guest before trying to configure the network. The debian cap does something similar https://github.com/hashicorp/vagrant/blob/master/plugins/guests/debian/cap/configure_networks.rb#L21-L32.

The guest inspection for this is already built out here https://github.com/hashicorp/vagrant/blob/master/lib/vagrant/util/guest_inspection.rb#L22. This could look something like:

```
if systemd_networkd?(comm)
  # do all the bits that use networkctl
else 
  # do all the bits that use netctl
end
```

Reply:
Nice catch. Could be useful for the transition period, too.

Reply:
Done in https://github.com/hashicorp/vagrant/commit/a86482b0176fcc395ca40217a16217c8246040ec, though see [my comment](https://github.com/hashicorp/vagrant/pull/11400#issuecomment-635703434) about the two failing unit tests. I'm not familiar enough with Ruby basics to know how to import the `systemd_networkd?` function.